### PR TITLE
plan, expression: remove useless error return

### PR DESCRIPTION
**Pull request by hanfei1991**

### CODE REVIEW

File: plan/util.go [hunk 1]
```diff
@@ -45,3 +45,36 @@ func (a *AggregateFuncExtractor) Leave(n ast.Node) (ast.Node, bool) {
 	}
 	return n, true
 }
+
+func replaceChild(p, child, replace Plan) {
```
Comment (XuHuaiyu) [inline plan/util.go]: add a comment for this.

File: expression/expression.go [hunk 2]
```diff
@@ -215,35 +215,25 @@ func SplitDNFItems(onExpr Expression) []Expression {
...
-				return nil, errors.Trace(err)
-			}
-		}
-		newFunc, err := NewFunction(ctx, x.FuncName.L, types.NewFieldType(mysql.TypeTiny), args...)
-		if err != nil {
-			return nil, errors.Trace(err)
+			args[i] = EvaluateExprWithNull(ctx, schema, arg)
 		}
-		return newFunc, nil
+		return NewFunctionInternal(ctx, x.FuncName.L, types.NewFieldType(mysql.TypeTiny), args...)
```
Comment (zz-jason) [inline expression/expression.go]: why change `NewFunction` to `NewFunctionInternal` ?

Comment (hanfei1991) [reply expression/expression.go]: It's impossible to report any error. This has been checked in expression_rewriter.

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**